Sources/MapboxMaps/Foundation/MapViewDependencyProvider.swift

```diff
@@ -244,21 +246,28 @@ internal final class MapViewDependencyProvider: MapViewDependencyProviderProtoco
 
     internal func makeLocationManager(locationProducer: LocationProducerProtocol,
                                       interpolatedLocationProducer: InterpolatedLocationProducerProtocol,
-                                      style: StyleProtocol) -> LocationManager {
+                                      style: StyleProtocol,
+                                      mapboxMap: MapboxMapProtocol,
+                                      displayLinkCoordinator: DisplayLinkCoordinator) -> LocationManager {
         let puckManager = PuckManager(
-            puck2DProvider: { configuration in
-                Puck2D(
+            puck2DProvider: { [weak displayLinkCoordinator] configuration in
+                guard let displayLinkCoordinator = displayLinkCoordinator else {
+                    fatalError("DisplayLinkCoordinator must be present when creating a 2D puck")
```

Comment:
why can't this be nil when creating a 2D puck? The puck is holding a weak reference to the coordinator anyway, so even if the coordinator is nil-checked when initialising, but there is still a chance that it will be deallocated when the puck becomes active, right?

Reply:
I saw this in AnnotationOrchestrator(namely `makePointAnnotationManager(id:layerPosition:)`). The idea is that the displayLinkCoordinator is a hard dependency of this component, but because of the reference cycle issue we have to use it as a weak dependency. Thus we can't rely on the compiler validation anymore and have to resort to checking if it exists in runtime.